samcli/commands/deploy/core/options.py

```diff
@@ -0,0 +1,87 @@
+"""
+Deploy Command Options related Datastructures for formatting.
+"""
+from typing import Dict, List
+
+from samcli.cli.row_modifiers import RowDefinition
+
+# The ordering of the option lists matter, they are the order in which options will be displayed.
+
+REQUIRED_OPTIONS: List[str] = ["stack_name", "capabilities", "resolve_s3"]
+
+# Can be used instead of the options in the first list
+INTERACTIVE_OPTIONS: List[str] = ["guided"]
+
+AWS_CREDENTIAL_OPTION_NAMES: List[str] = ["region", "profile"]
+
+INFRASTRUCTURE_OPTION_NAMES: List[str] = [
+    "parameter_overrides",
+    "s3_bucket",
+    "s3_prefix",
+    "resolve_image_repos" "image_repository",
+    "image_repositories",
+    "role_arn",
+    "kms_key_id",
+    "notification_arns",
+    "tags",
+    "metadata",
+]
+
+DEPLOYMENT_OPTIONS: List[str] = [
+    "no_execute_changeset",
+    "fail_on_empty_changeset",
+    "confirm_changeset",
+    "disable_rollback",
+    "on_failure",
+    "force_upload",
+]
+
+CONFIGURATION_OPTION_NAMES: List[str] = ["config_env", "config_file"]
+
+ADDITIONAL_OPTIONS: List[str] = [
+    "no_progressbar" "base_dir",
+    "signing_profiles",
+    "template_file",
+]
+
+OTHER_OPTIONS: List[str] = ["debug", "help"]
+
+ALL_OPTIONS: List[str] = (
```

Comment:
is there a way to test that we do not miss adding new command options here?

Reply:
Wrote a test for it!